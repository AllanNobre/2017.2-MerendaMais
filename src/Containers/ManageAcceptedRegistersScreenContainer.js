import { connect } from 'react-redux';
import ManageAcceptedRegistersScreen from '../screens/ManageAcceptedRegistersScreen';
import { asyncGetCounselorFromGroup } from '../actions/listActions';
import { disableCounselor, asyncAcceptCounselor } from '../actions/ManagerRegisterActions';

const mapStateToProps = state => (
  {
    listOfCounselorsInAGroup: state.list.listOfCounselorsInAGroup,
    counselor: state.counselor,
  }
);

const mapDispatchToProps = dispatch => (
  {
    asyncGetCounselorFromGroup: (CAE, CPF) => dispatch(asyncGetCounselorFromGroup(CAE, CPF)),
    disableCounselor: (counselor, codGroup) =>
      dispatch(disableCounselor(counselor, codGroup)),
    asyncAcceptCounselor: userData => dispatch(asyncAcceptCounselor(userData)),
  }
);

const ManageAcceptedRegistersScreenContainer =
  connect(mapStateToProps, mapDispatchToProps)(ManageAcceptedRegistersScreen);

export default ManageAcceptedRegistersScreenContainer;
