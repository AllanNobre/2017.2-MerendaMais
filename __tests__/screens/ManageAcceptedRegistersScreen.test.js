import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ManageAcceptedRegistersScreenContainer from '../../src/Containers/manageCounselors/ManageAcceptedRegistersScreenContainer';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {
  counselor: {
    token: 1,
    profile: {
      cpf: '11111111111',
      codGroup: '0',
      CAE: 'RO',
    },
  },
  list: {
    listOfCounselorsInAGroup: [{ nome: 'Joao' }],
  },
  asyncGetCounselorFromGroup: 12312321,
};

const store = mockStore(initialState);

describe('Testing ManageAcceptedRegistersScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <ManageAcceptedRegistersScreenContainer />,
      { context: { store } },
    );
    expect(wrapper).toMatchSnapshot();
  });
});
