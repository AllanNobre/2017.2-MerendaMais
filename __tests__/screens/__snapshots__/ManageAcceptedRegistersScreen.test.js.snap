// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ManageAcceptedRegistersScreen renders as expected 1`] = `
<ManageAcceptedRegistersScreen
  asyncAcceptCounselor={[Function]}
  asyncGetCounselorFromGroup={[Function]}
  counselor={
    Object {
      "profile": Object {
        "CAE": "RO",
        "codGroup": "0",
        "cpf": "11111111111",
      },
      "token": 1,
    }
  }
  disableCounselor={[Function]}
/>
`;
